en/resources/tutorials/codablerouting.md

```diff
@@ -0,0 +1,172 @@
+---
+### TRANSLATION INSTRUCTIONS FOR THIS SECTION:
+### TRANSLATE THE VALUE OF THE title ATTRIBUTE AND UPDATE THE VALUE OF THE lang ATTRIBUTE.
+### DO NOT CHANGE ANY OTHER TEXT.
+layout: page
+title: Codable Routing
+menu: resources
+lang: en
+redirect_from: "/resources/codablerouting.html"
+### END HEADER BLOCK - BEGIN GENERAL TRANSLATION
+---
+
+<div class="titleBlock">
+  <h1>Codable Routing</h1>
+  <p>How to use Codable Routing in Kitura</p>
+</div>
+
+
+
+## Introduction 
+Kitura 2.0 introduces “Codable Routing”. Here the the router handlers are much like normal functions you might define elsewhere in your code: they take struct or class types as parameters, and respond with stuct or class types via a completion handler, with just an additional requirement that those types implement [Codable](https://developer.apple.com/documentation/swift/codable) from Swift 4 (hence the name).
+
+Codable Routing isn’t suitable for every use case and scenario, so Raw Routing is still available where you need the power and flexibility, but it is perfect for building REST APIs such as you might want to do to build a Backend For Frontend (BFF) for an iOS app.
+
+## Building a Create API with a Codable Route for POST requests
+
+Building a Codable Route for a POST API to implement storing a To Do item requires only 4 lines of code. The first step is to register a handler for POST requests on “/todos”:
+
+```swift
+router.post("/todos", handler: storeHandler)
+```
+
+and then provide an implementation of the storeHandler: 
+
+```swift
+func storeHandler(todo: ToDo, completion: (ToDo?, RequestError?) -> Void ) -> Void {
+    todoStore.append(todo)
+    completion(todo, nil)
+}
+```
+
+Here the store handler accepts a todo item of type ToDo and responds using an asynchronous completion handler. The handler then stores the received todo item in the todoStore and returns with the created todo item in the case of success, and an Error in the the case of failure.
+
+The input parameter and the response parameter in completion handler can be any Swift type you want, as long as it conforms to Codable. For example, we could use the following as our ToDo type:
+
+```swift
+public struct ToDo : Codable {
+    public var title: String        // title of the todo item
+    public var order: Int           // order to display the todo item
+    public var completed: Bool      // has the item been completed
+}
+```
+
+At this point the server has stored the item, but the client has no way of referring to that specific item on the server if it wants to get that item, update it, or delete it. In order to do that, some kind of unique identifier is required.
+
+Returning an identifier to the client for the stored data is typically done in one of two ways:
```

Comment:
This is kind of nitpicking but maybe having an ordered list isn't the best way of suggesting options? The numbers (1, 2) suggests an order of steps rather than two individual options that stand alone. Maybe replace with bullet points or similar? (Or Option 1, Option 2)